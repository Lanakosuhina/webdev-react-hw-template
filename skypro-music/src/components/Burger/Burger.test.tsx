import renderer from "react-test-renderer";
import { Burger } from ".";

describe('Компонент бургер-меню', () => {
  const mockOnClick = jest.fn();

  it('корректно рендериться', () => {
    const component = renderer.create(<Burger
      onClick={mockOnClick}
    />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('корректно вызывает функцию принажатии', () => {
    const component = renderer.create(<Burger
      onClick={mockOnClick}
    />)

    let button = component.root.findByType("button"); 
    button.props.onClick();
    expect(mockOnClick).toHaveBeenCalled();
  });
});