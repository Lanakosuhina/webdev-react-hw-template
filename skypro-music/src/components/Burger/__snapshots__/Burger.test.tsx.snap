// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент бургер-меню корректно рендериться 1`] = `
<button
  className="navBurger"
  onClick={[MockFunction]}
>
  <span
    className="burgerLine"
  />
  <span
    className="burgerLine"
  />
  <span
    className="burgerLine"
  />
</button>
`;
